antarest/study/storage/variantstudy/model/command/remove_renewables_cluster.py

```diff
@@ -29,116 +30,95 @@ def __init__(self, **data: Any) -> None:
             **data,
         )
 
-    def _remove_renewables_cluster(
-        self, study_data_config: FileStudyTreeConfig
-    ) -> None:
-        study_data_config.areas[self.area_id].renewables = [
-            cluster
-            for cluster in study_data_config.areas[self.area_id].renewables
-            if cluster.id != self.cluster_id.lower()
-        ]
-
-        remove_area_cluster_from_binding_constraints(
-            study_data_config, self.area_id, self.cluster_id
-        )
-
     def _apply_config(
         self, study_data: FileStudyTreeConfig
     ) -> Tuple[CommandOutput, Dict[str, Any]]:
+        """
+        Applies configuration changes to the study data: remove the renewable clusters from the storages list.
+
+        Args:
+            study_data: The study data configuration.
+
+        Returns:
+            A tuple containing the command output and a dictionary of extra data.
+            On success, the dictionary is empty.
+        """
+        # Search the Area in the configuration
         if self.area_id not in study_data.areas:
             return (
                 CommandOutput(
                     status=False,
-                    message=f"Area '{self.area_id}' does not exist",
+                    message=(
+                        f"Area '{self.area_id}' does not exist"
+                        f" in the study configuration."
+                    ),
                 ),
-                dict(),
+                {},
             )
+        area: Area = study_data.areas[self.area_id]
 
-        if (
-            len(
-                [
-                    cluster
-                    for cluster in study_data.areas[self.area_id].renewables
-                    if cluster.id == self.cluster_id
-                ]
-            )
-            == 0
-        ):
+        # Search the Renewable cluster in the area
+        for renewable in area.renewables:
+            if renewable.id == self.cluster_id:
+                break
+        else:
             return (
                 CommandOutput(
                     status=False,
-                    message=f"Renewables cluster '{self.cluster_id}' does not exist",
+                    message=(
+                        f"Renewable cluster '{self.cluster_id}' does not exist"
+                        f" in the area '{self.area_id}'."
+                    ),
                 ),
-                dict(),
+                {},
             )
```

Comment:
`get` method to search for the `area_id` in the `study_data.areas` would be a bit more concise 

```python
    # get the area  in the configuration
    area = study_data.areas.get(self.area_id)
    if not area:
        return (
            CommandOutput(
                status=False,
                message=f"Area '{self.area_id}' does not exist in the study configuration."
            ),
            {},
        )

    # get the renewable cluster in the area
    if not any(renewable.id == self.cluster_id for renewable in area.renewables):
        return (
            CommandOutput(
                status=False,
                message=f"Renewable cluster '{self.cluster_id}' does not exist in the area '{self.area_id}'."
            ),
            {},
        )
```

Reply:
Actually, that is not equivalent because I need to find the matching `renewable`.

Instead, I can write:

```python
# Search the Renewable cluster in the area
renewable = next(
    iter(
        renewable
        for renewable in area.renewables
        if renewable.id == self.cluster_id
    ),
    None,
)
if renewable is None:
    message = (
        f"Renewable cluster '{self.cluster_id}' does not exist"
        f" in the area '{self.area_id}'."
    )
    return CommandOutput(status=False, message=message), {}
```

…or use a comprehension list, but that is not easy to understand.